kuma/javascript/src/payments/subscription-form.jsx

```diff
@@ -72,55 +72,118 @@ export default function SubscriptionForm() {
+            fetch(SUBSCRIPTION_URL, {
+                method: 'POST',
+                body: JSON.stringify({
+                    stripe_token: token.current, // eslint-disable-line camelcase
+                }),
+                headers: {
+                    'X-CSRFToken': getCookie('csrftoken'),
+                    'Content-Type': 'application/json',
                 },
+            }).then((response) => {
+                if (response.ok) {
+                    window.location = `/${locale}/payments/thank-you/`;
+                } else {
+                    console.error(
+                        'error while creating subscription',
+                        response.statusText
+                    );
+                    setFormStep('server_error');
+                }
             });
-            stripeHandler.open();
-        });
-    }
+        }
 
-    function createSubscription() {
-        fetch(SUBSCRIPTION_URL, {
-            method: 'POST',
-            body: JSON.stringify({
-                stripe_token: token.current, // eslint-disable-line camelcase
-            }),
-            headers: {
-                'X-CSRFToken': getCookie('csrftoken'),
-                'Content-Type': 'application/json',
-            },
-        }).then((response) => {
-            if (response.ok) {
-                window.location = `/${locale}/payments/thank-you/`;
-            } else {
-                console.error(
-                    'error while creating subscription',
-                    response.statusText
+        if (stripeLoadingPromise && openStripeModal) {
+            setFormStep('stripe');
+            stripeLoadingPromise.then(() => {
+                const stripeHandler = window.StripeCheckout.configure({
+                    key: window.mdn.stripePublicKey,
+                    locale,
+                    name: 'MDN Web Docs',
+                    zipCode: true,
+                    currency: 'usd',
+                    amount: 500,
+                    email: userData ? userData.email : '',
+                    // token is only called if Stripe was able to successfully
+                    // create a token from the entered info
+                    token(response) {
+                        token.current = response.id;
+                        createSubscription();
+                    },
+                    closed() {
+                        setFormStep(token.current ? 'submitting' : 'initial');
+                    },
+                });
+                stripeHandler.open();
+            });
+        }
+    }, [stripeLoadingPromise, openStripeModal, userData, locale]);
+
+    function submitHandler(event) {
+        event.preventDefault();
+        // Not so fast! If you're not authenticated yet, trigger the
+        // authentication modal instead.
+        if (userData && userData.isAuthenticated) {
+            setOpenStripeModal(true);
+        } else {
+            try {
+                sessionStorage.setItem(
```

Comment:
if you do extract the getter (as recommended above), I might also extract that right next to it which has the nice benefit of grouping serialization and deserialization.